sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/utils/JdbcUtils.java

```diff
@@ -0,0 +1,119 @@
+package org.apache.beam.sdk.schemas.utils;
+
+import org.apache.beam.sdk.schemas.LogicalTypes;
+import org.apache.beam.sdk.schemas.Schema;
+import org.joda.time.Instant;
+
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+import java.sql.Types;
+import java.util.HashMap;
+import java.util.Map;
+
+public class JdbcUtils {
+
+    private static final Map<Integer, String > RS_TYPE_MAPPING = new HashMap<>();
+
+    /**
+     *
+     */
+    static {
+        RS_TYPE_MAPPING.put(Types.BIT, "BIT");
+        RS_TYPE_MAPPING.put(Types.TINYINT, "TINYINT");
+        RS_TYPE_MAPPING.put(Types.BIGINT, "BIGINT");
+        RS_TYPE_MAPPING.put(Types.LONGVARBINARY, "LONGVARBINARY");
+        RS_TYPE_MAPPING.put(Types.VARBINARY, "VARBINARY");
+        RS_TYPE_MAPPING.put(Types.BINARY, "BINARY");
+        RS_TYPE_MAPPING.put(Types.LONGVARCHAR, "LONGVARCHAR");
+        RS_TYPE_MAPPING.put(Types.NULL, "NULL");
+        RS_TYPE_MAPPING.put(Types.CHAR, "CHAR");
+        RS_TYPE_MAPPING.put(Types.NUMERIC, "NUMERIC");
+        RS_TYPE_MAPPING.put(Types.DECIMAL, "DECIMAL");
+        RS_TYPE_MAPPING.put(Types.INTEGER, "INTEGER");
+        RS_TYPE_MAPPING.put(Types.SMALLINT, "SMALLINT");
+        RS_TYPE_MAPPING.put(Types.FLOAT, "FLOAT");
+        RS_TYPE_MAPPING.put(Types.REAL, "REAL");
+        RS_TYPE_MAPPING.put(Types.DOUBLE, "DOUBLE");
+        RS_TYPE_MAPPING.put(Types.VARCHAR, "VARCHAR");
+        RS_TYPE_MAPPING.put(Types.DATE, "DATE");
+        RS_TYPE_MAPPING.put(Types.TIME, "TIME");
+        RS_TYPE_MAPPING.put(Types.TIMESTAMP, "TIMESTAMP");
+        RS_TYPE_MAPPING.put(Types.OTHER, "OTHER");
+    }
+
+    private static Schema.FieldType fromFieldType(String typeName) { //List<TableFieldSchema> nestedFields
+        switch (typeName) {
+            case "VARCHAR":
+            case "CHAR":
+            case "LONGVARCHAR":
+            case "CLOB":
```

Comment:
Let's leave out the `CLOB` type because they can hold very large values and it might not be a good idea to treat them as strings.

Reply:
Done